sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveMetastoreCatalog.scala

```diff
@@ -689,6 +666,34 @@ private[hive] case class MetastoreRelation
     }
   )
 
+  def getHiveQlPartitions(filter: Option[String]): Seq[Partition] = {
```

Comment:
minor: I'd consider having this function also take an `Expression` or a `Seq[Expression]` so that the conversion from expressions to string is isolated from the rest of the code.